### ISSUE TITLE
XCode 10.2 iOS SDK 12.2 kJSTypeSymbol not handled

### ISSUE THREAD

**Issue by alexanderzhigulich** [LiquidPlayer/LiquidCore#96]
Hello,

Crash occurs when run simple script on LiquidCore v0.6.0

Script index.js from app bundle:
`
console.log('Hello from JS')
`

Swift code:
https://gist.github.com/alexanderzhigulich/489132e0d357ee0a63ba56b1257b6f7d
> if let path = Bundle.main.path(forResource: "index", ofType: "js"),
>     let service = LCMicroService(url: URL(fileURLWithPath: path), delegate: self) {
>     service.start()
> }

Crash:
https://gist.github.com/alexanderzhigulich/5e34856e8747fe941b0672ce24e0a5b7
> #0	0x0000000108dbf2c6 in __pthread_kill ()
> #1	0x0000000108e12bf1 in pthread_kill ()
> #2	0x0000000108b8201d in abort ()
> #3	0x0000000108b4aaa5 in __assert_rtn ()
> **#4	0x0000000104ad8476 in ValueImpl::New(ContextImpl const*, OpaqueJSValue const*, V82JSC_HeapObject::BaseMap*) at /LiquidCore/LiquidCoreiOS/LiquidCore/node-8.9.3/V82JSC/Value.cpp:92**
> #5	0x0000000104a7ea39 in v8::Symbol::New(v8::Isolate*, v8::Local<v8::String>) at /LiquidCore/LiquidCoreiOS/LiquidCore/node-8.9.3/V82JSC/Symbol.cpp:37
> #6	0x0000000104a7f685 in v8::Private::New(v8::Isolate*, v8::Local<v8::String>) at /LiquidCore/LiquidCoreiOS/LiquidCore/node-8.9.3/V82JSC/Symbol.cpp:159
> #7	0x00000001048a74f6 in node::IsolateData::IsolateData(v8::Isolate*, uv_loop_s*, unsigned int*) /LiquidCore/deps/node-8.9.3/src/env-inl.h:62
> #8	0x00000001048a0a1d in node::IsolateData::IsolateData(v8::Isolate*, uv_loop_s*, unsigned int*) at /LiquidCore/deps/node-8.9.3/src/env-inl.h:74
> #9	0x00000001049e2c10 in NodeInstance::StartInstance(uv_loop_s*, int, char const* const*, int, char const* const*) at /LiquidCore/LiquidCoreCommon/node/NodeInstance.cpp:819
> #10	0x00000001049e0102 in NodeInstance::StartInstance(int, char**) at /LiquidCore/LiquidCoreCommon/node/NodeInstance.cpp:964
> #11	0x00000001049dfe98 in NodeInstance::spawnedThread() at /LiquidCore/LiquidCoreCommon/node/NodeInstance.cpp:134
> #12	0x00000001049dfbe5 in NodeInstance::node_main_task(void*) at /LiquidCore/LiquidCoreCommon/node/NodeInstance.cpp:179
> #13	0x00000001049e7dd2 in decltype(std::__1::forward<void (*)(void*)>(fp)(std::__1::forward<void*>(fp0))) std::__1::__invoke<void (*)(void*), void*>(void (*&&)(void*), void*&&) at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4339
> #14	0x00000001049e7d1e in void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(void*), void*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(void*), void*>&, std::__1::__tuple_indices<2ul>) at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/thread:342
> #15	0x00000001049e7506 in void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (*)(void*), void*> >(void*) at /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/thread:352
> #16	0x0000000108e102eb in _pthread_body ()
> #17	0x0000000108e13249 in _pthread_start ()
> #18	0x0000000108e0f40d in thread_start ()

**Comment by ericwlange** [comment]
Thanks, I can confirm this bug.  It appears that the `kJSTypeSymbol` type was added in JavaScriptCore in the latest SDK.  Previously, symbols held the type `kJSTypeObject`.  I will patch this issue tomorrow.

**Comment by ericwlange** [comment]
To patch this issue, in `LiquidCore/LiquidCoreiOS/LiquidCore/node-8.9.3/V82JSC/Value.cpp`:

Add the lines:

```c++
            case kJSTypeSymbol: {
                map = isolateimpl->m_symbol_map;
                break;
            }
```

right before:
```c++
            default:
                break;
```

I can confirm this fixes the issue.  I will push out a version 0.6.1 once I have gone through testing.

**Comment by alexanderzhigulich** [comment]
Awesome!
I also confirm it's working.

Thank you!

**Comment by ericwlange** [comment]
Reopening for others who find this until I have a chance to push 0.6.1.

**Comment by ericwlange** [comment]
This has now been fixed in 0.6.1.
